src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -504,9 +504,13 @@ private TypeSymbol GetTypeOfTypeRef(TypeReferenceHandle typeRef, out bool isNoPi
 
             if (tokenType == HandleKind.AssemblyReference)
             {
-                // TODO: Can refer to the containing assembly?
                 isNoPiaLocalType = false;
-                return LookupTopLevelTypeDefSymbol(Module.GetAssemblyReferenceIndexOrThrow((AssemblyReferenceHandle)tokenResolutionScope), ref fullName);
+                var assemblyRef = (AssemblyReferenceHandle)tokenResolutionScope;
```

Comment:
This sounds reasonable, but we might want to hold on doing this change for Update 3 because we haven't seen a failure like that and the bar is high. Perhaps should open a separate issue to track the work.

---
In reply to: [65832042](https://github.com/dotnet/roslyn/pull/11772#discussion_r65832042) [](ancestors = 65832042)